lib/ClangImporter/ImportName.cpp

```diff
@@ -1286,6 +1286,7 @@ NameImporter::considerAsyncImport(
 
   // Initializers cannot be 'async'.
   // FIXME: We might eventually allow this.
+  // TODO: should the restriction be lifted in ClangImporter?
```

Comment:
Hmm. I think you might need to implement the corresponding restriction on the Swift side to prevent `@objc` `async` initializers, e.g., make sure that we reject

```swift
class X: NSObject {
  @objc init() async { ... }
}
```

because we can no longer rely on the the `async` init to be rejected.